index.js

```diff
@@ -1,20 +1,259 @@
+    this.roomName = roomName;
+    this.roomDescription = roomDescription;
+    this.locked = locked;
+    this.item = item;
+    this.connections = connections;
+  }
+
+  //move method on class Room to move between rooms
+  move() {
+    if (this.connections.includes(playerObject.currentRoom)) {
+      console.log(this.connections);
+      //playerObject.currentRoom = this.roomName;//transitions to the current room
+      //return `You are in ${currentRoom}. \n ${this.roomDescription}`;
+    }
+  }
+}
+
+//initialize rooms to be used in the game
+//the initial room
+let main = new Room(
+  "Main Room", //roomName
+  `You are in the main room of the house. \n Surrounding you are three doors. \n Examine the doors for clues.`, //roomDescription
+  false, //locked true or false
+  "door", //item
+  ["The Enchanted Forest", "The Puzzle Room", "The Chasm"] //connections
+);
+//a room in the game where items are able to be picked up to advance in the game
+let enchanted = new Room(
+  "The Enchanted Forest", //roomName
+  `You have traveled through a portal and entered the enchanted forest dimension. \n 
+  From one of the tree hollows you spy a glint of light. Examine the tree for clues.`, //roomDescription
+  false, //locked true or false
+  "key", //item
+  "Main Room" //connections
+);
 
+//a room where the puzzle takes place in the game
+let puzzle = new Room(
+  "The Puzzle Room", //roomName
+  `You are now in the puzzle room. \n 
+    To pass into the next dimension you must enter a code to pass to the next room.`, //roomDescription
+  true, //locked true or false
+  "mushroom", //item
+  ["The Nimsesku Room", "Room #1"]
+);
+
+//a room with nothing in it
+let emptiness = new Room(
+  "The Emptiness", //roomName
+  `You have entered the emptiness. \n
+    There is nothing here but a sign.`, //roomDescription
+  false, // locked true or false
+  "sign", //item
+  ["The Abyss", "Room #1"] //connections
+);
+
+//a room with nothing in it that ends the game when you enter it
+let abyss = new Room(
+  "The Abyss", //roomName
+  `You have entered the abyss. \n
+    There is no turning back now. The wormhole has sucked you in. \n
+    The universe has imploded. Game over.`, //roomDescription
+  false, //locked true or false
+  undefined // no connecting rooms
+);
+
+//the room you enter that wins the game
+let nimsesku = new Room(
+  "The Nimsesku Room", //roomName
+  `You have found Nimsesku! \n
+    All is restored and you have saved the universe as we know it! \n
+    You have won the game!`, //roomDescription
+  false, //locked true or false
+  undefined // no connecting rooms
+);
+
+//lookup table for room class (looks up string version and spits out variable version allowing the variable version to be looked up)
+let lookupRoomTable = {
+  main: main,
+  enchanted: enchanted,
+  puzzle: puzzle,
+  emptiness: emptiness,
+  abyss: abyss,
+  nimsesku: nimsesku,
+};
+
+//state machine to move between the rooms
```

Comment:
I think you may have figured out you can implement your state machine in your Room objects themselves, if so and you're not going to use this feel free to delete it